clowder-framework/clowder2
# 904 allow folder renaming

Backend change:
- Added PATCH /datasets/{datasetId}/folders/{folderID} endpoint
- Added GET /datasets/{datasetId}/folders/{folderID} endpoint --> but not being used in the frontend yet
- Pytest added

Frontend:
- Rename folder capability
- Authwrapper wrap around the folder menu to only allow "owner", "editor" and "admin" to delete or rename folder

<img width="927" alt="Pasted Graphic 4" src="https://github.com/clowder-framework/clowder2/assets/13950475/49ecf94b-34aa-4f48-8bf5-fdae74130d3e">
<img width="784" alt="Rename Folder" src="https://github.com/clowder-framework/clowder2/assets/13950475/3fa0d174-deb4-4d51-902c-58e0ad807948">
<img width="950" alt="Pasted Graphic 5" src="https://github.com/clowder-framework/clowder2/assets/13950475/8420bfad-16b4-4990-9d90-50359f836bd8">

-----

To test:

1. Run the pytest
2. Create a few folders and nested folders
3. Rename, delete, add files see if everything works
4. Switch to public view to make sure "folder menu" disappeard
